### DISCUSSION TITLE
Invoice not updated after payment

### DISCUSSION THREAD

**Discussion by mvanderkruk** [pretix/pretix#3790] [Q&A]
I have set the invoice to be updated in the organizer settings: "Allow to update existing invoices" is ticked.
In the event Invoice settings I have set the Generate invoices to: "Automatically after payment or when required by payment method"
When I import the bank transfers and an order is partially paid or completely paid, the generated invoice still states an outstanding payment. The order is marked as fully paid in the order screen, buth the downloaded invoice still states: 
![image](https://github.com/pretix/pretix/assets/2444979/ba9e9f68-ec49-4be7-8232-379217baeda6)

I have to manually regenerate the invoice to show the invoice as being completely paid on the invoice.

We need this because of international orders which are sometimes not fully paid and we need the invoices to be generated with the outstanding amount on them.

Or am I doing something wrong?

**Comment by mvanderkruk** [comment]
Also upon further testing when an order is overpaid, it does not show on the invoice....

**Comment by raphaelm** [comment]
This is a misunderstanding, or at least the system is working as intended.

The system will **never ever** automatically change an invoice after it has issued. Invoice documents are supposed to be immutable (= once created, they can't change) in many countries. The **only** way an invoice may change after it is created is if a backend user actively presses the "Regenerate invoice" button, which – to prevent accidental usage in countries where it might be problematic – is now hidden by default unless "Allow to update existing invoices" is enabled.

**Reply by mvanderkruk** [reply]
Would it be possible to generate a new (up to date) invoice after (every) payment? 
As a customer I would like to have an invoice of what I still need to pay. Currently this information is only displayed on the order overview page.

**Reply by raphaelm** [reply]
Possible, sure, but at least the way invoices are treated in Germany it would be a horrible experience for any business user on both ends (seller and buyer) if we did this automatically, since with every payment the old invoice would be voided and a new invoice with a new date and number would be issued that needs to be properly recorded in the accounting department.
